server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -88,10 +88,13 @@ class LivyServer extends Logging {
         }
     }
 
+    // TODO Create a new class to pass variables from LivyServer to sessions and remove these
+    // internal LivyConfs.
     // Set formatted Spark and Scala version into livy configuration, this will be used by
     // session creation.
     livyConf.set(LIVY_SPARK_VERSION.key, formattedSparkVersion.productIterator.mkString("."))
-    livyConf.set(LIVY_SPARK_SCALA_VERSION.key,
+    // Integration test sets spark scala version.
+    livyConf.setIfMissing(LIVY_SPARK_SCALA_VERSION.key,
```

Comment:
This will introduce bug if user set `LIVY_SPARK_SCALA_VERSION` like `2.10.4` or `2.11.8`. We should format user's input to `major.minor` and reset to LivyConf to override it. Otherwise if user set "2.10.4", this will introduce issue in `InteractiveSession`.